### PR TITLE
[Purview - Account API] new version

### LINKED ISSUE #21056: [Purview - Account API] API Review
New API Review meeting has been requested.

**Service Name**: Microsoft Purview
**Review Created By**: Phani Kumar Vadali
**Review Date**: 10/26/2022 9:00 AM PT
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/21055
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: 

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=f1ae5340-f648-ed11-bba1-000d3a341bd5)

### PR DESCRIPTION
**Pull request by pvadali-az**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths: Added AccountPreferences endpoint. [here](https://github.com/Azure/azure-rest-api-specs/blob/456efbfc18ed08dd9a9e216f9e20923cb7b54774/specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json#L566-L666) is the block added to `account.json`

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

fix https://github.com/Azure/azure-rest-api-specs/issues/21056

### CODE REVIEW

File: specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json [hunk 1]
```diff
@@ -0,0 +1,1764 @@
...
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "ResourceSetRules_ListResourceSetRules": {
+            "$ref": "./examples/ResourceSetRules_ListResourceSetRules.json"
+          }
+        }
+      }
+    },
+    "/accountpreferences": {
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json:566]: ✅ DO use kebab-casing (preferred) or camel-casing for URL path segments. If the segment refers to a JSON field, use camel casing.

Existing API seems to be camelCase. Expect to follow it?

File: specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json [hunk 2]
```diff
@@ -0,0 +1,1764 @@
...
+    "Cobranding": {
+      "description": "Cobranding object.",
+      "type": "object",
+      "properties": {
+        "logo": {
+          "description": "logo",
+          "type": "string"
+        },
+        "color": {
+          "description": "color",
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json:1709]: I would expect more detail about what is expected. I saw "blue" in example, does it accept e.g. RGB#, or "green yellow"?

File: specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json [hunk 3]
```diff
@@ -0,0 +1,1764 @@
...
+        }
+      }
+    },
+    "/accountpreferences": {
+      "get": {
+        "tags": [
+          "AccountPreferences"
+        ],
+        "description": "Get Account Preferences",
+        "operationId": "AccountPreferences_GetAccountPreferences",
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json:572]: nit, it is a bit confusing to have plural "AccountPreferences" but this is actually not a collection. I've no good idea here, could be discussed in meeting.

File: specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json [hunk 4]
```diff
@@ -0,0 +1,1764 @@
...
+        }
+      }
+    },
+    "AccountPreferences": {
+      "description": "Account Preferences resource.",
+      "type": "object",
+      "properties": {
+        "cobranding": {
+          "$ref": "#/definitions/Cobranding"
+        }
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json:1697]: do we expect more, than current "cobranding"?

File: specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json (hunk 1 above)
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json]: changed to camelCase

File: specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json (hunk 2 above)
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json]: Added description and changed example accordingly

**Comment by pvadali-az** [comment]
> @microsoft-github-policy-service agree [company="{your company}"]

@microsoft-github-policy-service agree [company="{microsoft}"]

**Comment by pvadali-az** [comment]
> @pvadali-az please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> 
> Contributor License Agreement
> ### Contribution License Agreement
> This Contribution License Agreement (**“Agreement”**) is agreed to by the party signing below (**“You”**), and conveys certain license rights to Microsoft Corporation and its affiliates (“Microsoft”) for Your contributions to Microsoft open source projects. This Agreement is effective as of the latest signature date below.
> 
> 1. **Definitions**.
>    **“Code”** means the computer software code, whether in human-readable or machine-executable form,
>    that is delivered by You to Microsoft under this Agreement.
>    **“Project”** means any of the projects owned or managed by Microsoft and offered under a license
>    approved by the Open Source Initiative ([www.opensource.org](http://www.opensource.org)).
>    **“Submit”** is the act of uploading, submitting, transmitting, or distributing code or other content to any
>    Project, including but not limited to communication on electronic mailing lists, source code control
>    systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
>    discussing and improving that Project, but excluding communication that is conspicuously marked or
>    otherwise designated in writing by You as “Not a Submission.”
>    **“Submission”** means the Code and any other copyrightable material Submitted by You, including any
>    associated comments and documentation.
> 2. **Your Submission**. You must agree to the terms of this Agreement before making a Submission to any
>    Project. This Agreement covers any and all Submissions that You, now or in the future (except as
>    described in Section 4 below), Submit to any Project.
> 3. **Originality of Work**. You represent that each of Your Submissions is entirely Your original work.
>    Should You wish to Submit materials that are not Your original work, You may Submit them separately
>    to the Project if You (a) retain all copyright and license information that was in the materials as You
>    received them, (b) in the description accompanying Your Submission, include the phrase “Submission
>    containing materials of a third party:” followed by the names of the third party and any licenses or other
>    restrictions of which You are aware, and (c) follow any other instructions in the Project’s written
>    guidelines concerning Submissions.
> 4. **Your Employer**. References to “employer” in this Agreement include Your employer or anyone else
>    for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
>    Submission is made in the course of Your work for an employer or Your employer has intellectual
>    property rights in Your Submission by contract or applicable law, You must secure permission from Your
>    employer to make the Submission before signing this Agreement. In that case, the term “You” in this
>    Agreement will refer to You and the employer collectively. If You change employers in the future and
>    desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
>    and secure permission from the new employer before Submitting those Submissions.
> 5. **Licenses**.
> 
> * **Copyright License**. You grant Microsoft, and those who receive the Submission directly or
>   indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the
>   Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute
>   the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third
>   parties.
> * **Patent License**. You grant Microsoft, and those who receive the Submission directly or
>   indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under
>   Your patent claims that are necessarily infringed by the Submission or the combination of the
>   Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
>   import or otherwise dispose of the Submission alone or with the Project.
> * **Other Rights Reserved**. Each party reserves all rights not expressly granted in this Agreement.
>   No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
>   granted by implication, exhaustion, estoppel or otherwise.
> 
> 6. **Representations and Warranties**. You represent that You are legally entitled to grant the above
>    licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
>    have disclosed under Section 3). You represent that You have secured permission from Your employer to
>    make the Submission in cases where Your Submission is made in the course of Your work for Your
>    employer or Your employer has intellectual property rights in Your Submission by contract or applicable
>    law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
>    have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
>    You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
>    REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
>    EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
>    PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
>    NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
> 7. **Notice to Microsoft**. You agree to notify Microsoft in writing of any facts or circumstances of which
>    You later become aware that would make Your representations in this Agreement inaccurate in any
>    respect.
> 8. **Information about Submissions**. You agree that contributions to Projects and information about
>    contributions may be maintained indefinitely and disclosed publicly, including Your name and other
>    information that You submit with Your Submission.
> 9. **Governing Law/Jurisdiction**. This Agreement is governed by the laws of the State of Washington, and
>    the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
>    Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
>    exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
>    defenses of lack of personal jurisdiction and forum non-conveniens.
> 10. **Entire Agreement/Assignment**. This Agreement is the entire agreement between the parties, and
>     supersedes any and all prior agreements, understandings or communications, written or oral, between
>     the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.

**Comment by pvadali-az** [comment]
> > @pvadali-az please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> > ```
> > @microsoft-github-policy-service agree [company="{your company}"]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > > Options:
> > > 
> > > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > 
> > Contributor License Agreement
> > ### Contribution License Agreement
> > This Contribution License Agreement (**“Agreement”**) is agreed to by the party signing below (**“You”**), and conveys certain license rights to Microsoft Corporation and its affiliates (“Microsoft”) for Your contributions to Microsoft open source projects. This Agreement is effective as of the latest signature date below.
> > 
> > 1. **Definitions**.
> >    **“Code”** means the computer software code, whether in human-readable or machine-executable form,
> >    that is delivered by You to Microsoft under this Agreement.
> >    **“Project”** means any of the projects owned or managed by Microsoft and offered under a license
> >    approved by the Open Source Initiative ([www.opensource.org](http://www.opensource.org)).
> >    **“Submit”** is the act of uploading, submitting, transmitting, or distributing code or other content to any
> >    Project, including but not limited to communication on electronic mailing lists, source code control
> >    systems, and issue tracking systems that are managed by, or on behalf of, the Project for the purpose of
> >    discussing and improving that Project, but excluding communication that is conspicuously marked or
> >    otherwise designated in writing by You as “Not a Submission.”
> >    **“Submission”** means the Code and any other copyrightable material Submitted by You, including any
> >    associated comments and documentation.
> > 2. **Your Submission**. You must agree to the terms of this Agreement before making a Submission to any
> >    Project. This Agreement covers any and all Submissions that You, now or in the future (except as
> >    described in Section 4 below), Submit to any Project.
> > 3. **Originality of Work**. You represent that each of Your Submissions is entirely Your original work.
> >    Should You wish to Submit materials that are not Your original work, You may Submit them separately
> >    to the Project if You (a) retain all copyright and license information that was in the materials as You
> >    received them, (b) in the description accompanying Your Submission, include the phrase “Submission
> >    containing materials of a third party:” followed by the names of the third party and any licenses or other
> >    restrictions of which You are aware, and (c) follow any other instructions in the Project’s written
> >    guidelines concerning Submissions.
> > 4. **Your Employer**. References to “employer” in this Agreement include Your employer or anyone else
> >    for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your
> >    Submission is made in the course of Your work for an employer or Your employer has intellectual
> >    property rights in Your Submission by contract or applicable law, You must secure permission from Your
> >    employer to make the Submission before signing this Agreement. In that case, the term “You” in this
> >    Agreement will refer to You and the employer collectively. If You change employers in the future and
> >    desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement
> >    and secure permission from the new employer before Submitting those Submissions.
> > 5. **Licenses**.
> > 
> > 
> > * **Copyright License**. You grant Microsoft, and those who receive the Submission directly or
> >   indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license in the
> >   Submission to reproduce, prepare derivative works of, publicly display, publicly perform, and distribute
> >   the Submission and such derivative works, and to sublicense any or all of the foregoing rights to third
> >   parties.
> > * **Patent License**. You grant Microsoft, and those who receive the Submission directly or
> >   indirectly from Microsoft, a perpetual, worldwide, non-exclusive, royalty-free, irrevocable license under
> >   Your patent claims that are necessarily infringed by the Submission or the combination of the
> >   Submission with the Project to which it was Submitted to make, have made, use, offer to sell, sell and
> >   import or otherwise dispose of the Submission alone or with the Project.
> > * **Other Rights Reserved**. Each party reserves all rights not expressly granted in this Agreement.
> >   No additional licenses or rights whatsoever (including, without limitation, any implied licenses) are
> >   granted by implication, exhaustion, estoppel or otherwise.
> > 
> > 
> > 6. **Representations and Warranties**. You represent that You are legally entitled to grant the above
> >    licenses. You represent that each of Your Submissions is entirely Your original work (except as You may
> >    have disclosed under Section 3). You represent that You have secured permission from Your employer to
> >    make the Submission in cases where Your Submission is made in the course of Your work for Your
> >    employer or Your employer has intellectual property rights in Your Submission by contract or applicable
> >    law. If You are signing this Agreement on behalf of Your employer, You represent and warrant that You
> >    have the necessary authority to bind the listed employer to the obligations contained in this Agreement.
> >    You are not expected to provide support for Your Submission, unless You choose to do so. UNLESS
> >    REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING, AND EXCEPT FOR THE WARRANTIES
> >    EXPRESSLY STATED IN SECTIONS 3, 4, AND 6, THE SUBMISSION PROVIDED UNDER THIS AGREEMENT IS
> >    PROVIDED WITHOUT WARRANTY OF ANY KIND, INCLUDING, BUT NOT LIMITED TO, ANY WARRANTY OF
> >    NONINFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
> > 7. **Notice to Microsoft**. You agree to notify Microsoft in writing of any facts or circumstances of which
> >    You later become aware that would make Your representations in this Agreement inaccurate in any
> >    respect.
> > 8. **Information about Submissions**. You agree that contributions to Projects and information about
> >    contributions may be maintained indefinitely and disclosed publicly, including Your name and other
> >    information that You submit with Your Submission.
> > 9. **Governing Law/Jurisdiction**. This Agreement is governed by the laws of the State of Washington, and
> >    the parties consent to exclusive jurisdiction and venue in the federal courts sitting in King County,
> >    Washington, unless no federal subject matter jurisdiction exists, in which case the parties consent to
> >    exclusive jurisdiction and venue in the Superior Court of King County, Washington. The parties waive all
> >    defenses of lack of personal jurisdiction and forum non-conveniens.
> > 10. **Entire Agreement/Assignment**. This Agreement is the entire agreement between the parties, and
> >     supersedes any and all prior agreements, understandings or communications, written or oral, between
> >     the parties relating to the subject matter hereof. This Agreement may be assigned by Microsoft.

@microsoft-github-policy-service agree company="Microsoft"

File: specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json (hunk 3 above)
Comment (markweitzel) [reply specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2022-10-01-preview/account.json:572]: @weidongxu-microsoft -- Agree it's a bit confusing. Looking at the model, this feels like "settings", e.g. "logo", "color"...
I'd expect something like, `/account/{accountId}/preferences`. In this case, even if it's not a collection, the idea that I'd get an object back that represents my account preferences makes sense.

**Pull request closed without merging**